libavcodec/vaapi_decode.c

```diff
@@ -564,10 +586,32 @@ static int vaapi_decode_make_config(AVCodecContext *avctx,
 
     if (frames_ref) {
         AVHWFramesContext *frames = (AVHWFramesContext *)frames_ref->data;
+        AVVAAPIFramesContext *avfc = frames->hwctx;
 
         frames->format = AV_PIX_FMT_VAAPI;
         frames->width = avctx->coded_width;
         frames->height = avctx->coded_height;
+        avfc->enable_sub_frame = (avctx->export_side_data & AV_CODEC_EXPORT_DATA_SUB_FRAME) && support_sfc;
```

Comment:
I think "avctx->export_side_data & AV_CODEC_EXPORT_DATA_SUB_FRAME" is redundant.